source/MaterialXCore/Node.cpp

```diff
@@ -197,7 +197,7 @@ OutputPtr Node::getNodeDefOutput(ElementPtr connectingElement)
 vector<PortElementPtr> Node::getDownstreamPorts() const
 {
     vector<PortElementPtr> downstreamPorts;
-    for (PortElementPtr port : getDocument()->getMatchingPorts(getName()))
+    for (PortElementPtr port : getDocument()->getMatchingPorts(getQualifiedName(getName())))
```

Comment:
This make it consistent with cache rebuild which stores qualified names. Every setAttribute which includes setting the namespace appears to invalidate the cache.

Reply:
Also @ZapAndersson, can you add a test for getMatchingPorts() with a namespaced example and also add in your flatten example. 

I thought I had added one in before but I think it got removed with the Syntax directory was cleaned up @jstone-lucasfilm ? Anyways this is what I had which I'm unsure if it matches what I put in here: https://github.com/autodesk-forks/MaterialX/pull/1173, but seems like this didn't cover everything as it ran validate() on the flattend graph without issue.

Reply:
I can try. I just might a be a bit too stupid to do that, practically :P

Reply:
For the first, there is already a test which uses this getDownstreamPorts() so that can just be enhanced so that
the it tests a graph with a namespace connected to something upstream.

For the second, the flattening test got removed by accident since it referenced a file in the Syntax area which has been removed. The original test file could have missed testing your case or didn't include your config, but that a test file can be added back in, and the flatten test restored to check for valid connections after flattening.
See: https://github.com/AcademySoftwareFoundation/MaterialX/commit/eb813c3ba7bf02df10f301b1fcca1f44d62cc734#diff-0f847260949f9961b33877230dbb8dbb6e4140fae69640748d5c582f3f1445c7L154
or the ADSK check-in as mentioned.

Reply:
I have a PR https://github.com/AcademySoftwareFoundation/MaterialX/pull/892 opened that brings in Flatten test from ADSK contrib

Reply:
@ashwinbhat okay, so... should I wait with this until yours is checked in and then add my test to the existing flattening test or... ?

Reply:
As #892 is a mix of updates why not take the flatten changes (including the predicate option addition) + test out from there and move it here. Then @ZapAndersson  you can update the test file.

Reply:
That works for me, my change is obviouly a quarter of a line. Do you wanna suck in my change to your PR, @ashwinbhat ?
(I can fix tests after)?

Reply:
I'll update #892 to remove the flatten changes.

Reply:
@kwokcb I think this is ready to go in :)